src/__experimental__/components/date/date.component.js

```diff
@@ -0,0 +1,259 @@
+      this.updateDatePickerValue(this.props.value);
+    }
+
+    this.setState({ isDatePickerOpen: true });
+  };
+
+  closeDatePicker = () => {
+    this._document.removeEventListener('click', this.closeDatePicker);
+    this.setState((prevState) => {
+      return {
+        visibleValue: this.formatVisibleValue(prevState.selectedDate),
+        isDatePickerOpen: false
+      };
+    });
+  };
+
+  handleDateSelect = (selectedDate, modifiers) => {
+    const stringDateIso = DateHelper.formatDateString(selectedDate, isoDateFormat);
+
+    if (modifiers.disabled) return;
+
+    this.blockBlur();
+    this.closeDatePicker();
+    this.updateVisibleValue(selectedDate);
+    this.emitOnChangeCallback(stringDateIso);
+  };
+
+  updateVisibleValue = (date) => {
+    const formattedDateString = this.formatVisibleValue(date);
+    this.setState({ visibleValue: formattedDateString });
+  };
+
+  formatVisibleValue(value) {
+    // Don't sanitize so it accepts the hidden format (with dash separators)
+    return DateHelper.formatValue(value || today, getVisibleFormat(), {
+      formats: isoDateFormat,
+      sanitize: false
+    });
+  }
+
+  handleVisibleInputChange = (ev) => {
+    const dateWithSlashes = DateHelper.sanitizeDateInput(ev.target.value);
+    const isValidDate = DateHelper.isValidDate(dateWithSlashes);
+    const isoDateString = DateHelper.formatDateString(dateWithSlashes, isoDateFormat);
+
+    // Updates the hidden value after first formatting to default hidden format
+    if (isValidDate) {
+      this.updateDatePickerValue(dateWithSlashes);
+      this.emitOnChangeCallback(isoDateString);
+    }
+
+    this.setState({ visibleValue: ev.target.value });
+  };
+
+  updateDatePickerValue = (newValue) => {
+    const isoDate = convertToIsoDate(newValue);
+
+    this.setState({ selectedDate: isoDate });
+  };
+
+  emitOnChangeCallback = (stringDate) => {
+    this.props.onChange({ target: { value: stringDate } });
+  };
+
+  renderDatePicker = (passedProps) => {
+    const datePickerProps = {
+      inputElement: this.input && this.input.parentElement,
+      selectedDate: this.state.selectedDate || convertToIsoDate(this.state.visibleValue),
+      handleDateSelect: this.handleDateSelect,
+      ...passedProps
+    };
+    console.log('rendered');
+
+    return <DatePicker { ...datePickerProps } />;
+  }
+
+  render() {
+    const isComponentActive = !this.props.disabled && !this.props.readOnly;
+    const { minDate, maxDate, ...inputProps } = this.props;
+    let events = {};
+    delete inputProps.autoFocus;
+
+    if (isComponentActive) {
+      events = {
+        onBlur: this.handleBlur,
+        onChange: this.handleVisibleInputChange,
+        onFocus: this.handleFocus,
+        onKeyDown: this.handleKeyDown
+      };
+    }
+
+    return (
+      <StyledDateInput
+        onClick={ stopClickPropagation }
+        onKeyDown={ this.handleKeyDown }
+        role='presentation'
+      >
+        <InputDecoratorBridge
```

Comment:
Do we want to remove the Bridge with this work?

Reply:
we need validations in this component

Reply:
Validations are almost finished https://github.com/Sage/carbon/pull/2002

Your Date component will inherit these automatically, we should avoid any further reliance on `InputDecoratorBridge`

Reply:
(same applies to `Number`, which is already in master - we will need to remove it)

Reply:
We can QA the validations as part of the validations ticket, they should work across all input components.

Reply:
ok, I'll swap it for the Textbox then